testing_app/README.md

```diff
@@ -0,0 +1,33 @@
+# testing_app
+
+A Sample app that shows different types of testing in Flutter.
+
+This particular sample uses the [Provider][] package but any other state management approach
+would do.
+
+[Provider]: https://pub.dev/packages/provider
+
+This sample is currently being built. Not all testing types are covered yet.
+
+## Goals for this sample
+Show how to perform:
+* Widget Testing,
+* Flutter Driver(Integration) Testing,
+* Performance Testing, and
+* State Management Testing using the [Provider][] package.
+
+### How to run tests
+* Widget Tests:
+  * To run on physical device or emulator:
+    * Navigate to the project's root folder using command line.
+    * Run ```flutter run test/<file_path>```
```

Comment:
I've added instructions to run the tests. Will add more as soon as we add new testing types. Please let me know if you think it's unnecessary.